draft-ietf-teep-architecture.md

```diff
@@ -1347,6 +1347,33 @@ still a threat present in allowing a potentially compromised TAM to install
 any TA it wants; for example, a known-compromised TA or an explicitly
 malicious TA.
 
+The device manufacturer may want to include a notification support that can allow
+Device Administrators be notified for incidents upon detection with some mitigation 
+methods as suggested above.
+
+## Hostile TAM
+
+There are also threats of hostile or abusive TAMs where a TAM turns to act out of 
+expectation of Device Administrators, for example, pushing out TAs that contain 
+some data collection or use users' device resources for distributed jobs for a TAM. 
+To mitigate this threat, TEEs and Device Administrators have several options for 
+detecting and mitigating a hostile TAM:
+
+1. ACLs: Apply an ACL to TAMs, making explicit which TAs the TAM is permitted to install. 
+This limits the scope of a hostile TAM's impact.
```

Comment:
I'm not sure it really limits it.  If you have a way of managing such an ACL by knowing things like the hash of the TA, then you really are acting like a TAM.  Besides, as shown in the "Keys and Certificate Types" section, it's the Code Signing keys that control what TA's can be installed, not the TAM keys per se.   In other words, no point in allowing installation of code that fails the code signing check.   (As opposed to personalization data only components, for which the text says "Note that Personalization Data is not included in the table above.")

Reply:
It is hard to implement, I agree. We already want to use Trust Anchor as the guard. Individual abusive TA can be removed but nothing prevents a TAM to reinstall it :) So the TAM needs to be blocked if it repeats, where the other mitigation methods #2,3,4 will do.

Let me remove this for now. The same should apply to the "Compromised TAM" section where ACLs is possible but not likely be effective.

Reply:
My opinion: a hostile TAM vs a compromised TAM is the same case.   The Device Administrator is responsible for choosing what TAM to trust with their device.   Once they trust one, it going bad is just like Device Administrator's employee going bad.  You use social/contractual enforcement to deal with it, it's not a technical issue.

Reply:
I have removed ACL. But still keep them separate. The "Apply ACL in TAM" is a mitigation for Compromised TAM by a TAM provider itself. Do you feel strongly that we merge the two back to one but leave the paragraph content the same under the title "Compromised TAM" or "Compromised or Hostile TAM"?

Hostile is an active behavior similar to privacy invasion by ads companies.

Reply:
A TEEP Agent and Device Administrator cannot distinguish (in any technical way) between a "hostile" TAM and a "compromised" TAM and so have to treat them as the same case.   The Device Administrator picks the TAM(s) and implicitly trusts them.   If there are multiple, I do think it's ok to let them manage different things (e.g., trust anchor store, TEEP agent code, secure firmware/OS, etc.) so ACL is ok if expressed appropriately in text.

Reply:
The idea behind my original "apply an ACL" comment was this. Imagine a scenario like installing signed software. You may have both Vendor A and Vendor B in your root store, but Vendor B should not be able to sign Vendor A's software and Vendor A should not be able to sign Vendor B's software.

Similarly in TA space, a TAM may be trusted to manage some content but not other content. I can easily imagine a scenario where two large companies both insist on operating their own TAMs. They could include banks or technology companies, etc. Each of these TAMs then should have strict limits placed on *which TAs* they are allowed to install. Now, when I say "which TAs," it might be that a better abstraction is "which trust domains they may install TAs into."

Reply:
Well in particular you could in theory say that if you have multiple TAMs, that only the TAM that installed a given trusted component can update it.   However, I think that would unnecessarily mix two different things from [Figure 4](https://www.ietf.org/archive/id/draft-ietf-teep-architecture-18.html#name-keys-and-certificate-types).  Trusted Component install is based on code signing keys not TAM keys.   So to install anything, you need a manifest signed by a code signing key in the TEE's code signing trust anchor store.   That store is separate from the TEEP Agent's store of TAM keys, as the figure shows.

For installation of _new_ trusted components, I think Brendan is right about "which trust domains they may install TAs into".   However per issues #7 and #62 the WG consensus was to remove all mention of security domains from the base architecture.  Similarly in SUIT, trust domains were separated out into an extension.   So it's difficult to reference them now, and I'd prefer some wording that does not pull that concept back in now.

Reply:
While my original wording was a little fuzzy, given that I'm not specifying **how** an ACL can specify which TAs a TAM is permitted to install, it does deal with the question of limiting damage without reintroducing trust domains. I think we should probably stick with that, leaving exactly what the ACL does up to implementors How about this:

ACLs: Make explicit which TAs a TAM is permitted to install. This limits the scope for damage. ACLs could restrict which path prefixes a TAM can use, which TA Components they are permitted to access, which TA Signers must have signed the TAs they install, etc.

Reply:
Add that a compromised TAM cannot install arbitrary TA unless such a TA is signed by a trusted signer first. I have adjusted the threat description to reflect this.